fobnail-attester.c

```diff
@@ -0,0 +1,158 @@
+/**
+ * Some LICENSE
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <arpa/inet.h>
+#include <coap3/coap.h>
+#include <signal.h>
+#include <stdbool.h>
+
+static bool quit = false;
+static const char LISTEN_ADDRESS[] = "0.0.0.0";
+static unsigned int port = COAP_DEFAULT_PORT; /* default port 5683 */
+
+static void handle_sigint(int signum)
+{
+	quit = true;
+}
+
+static void coap_attest_handler(struct coap_resource_t* resource, struct coap_session_t* session,
+				const struct coap_pdu_t* in, const struct coap_string_t* query,
+				struct coap_pdu_t* out)
+{
+	int ret;
+	char *res_buf = "Response from server.\n";
+	size_t res_buf_len = strlen(res_buf);
+
+	printf("Received message.\n");
+
+	/* prepare and send response */
+	ret = coap_add_data_large_response(resource,
```

Comment:
@0xDen Add following call before `coap_add_data_large_response`
```
coap_pdu_set_code(out, COAP_RESPONSE_CODE_CONTENT);
```
Otherwise server sends some random trash (or doesn't send anything at all).
```
Creating CoAP server endpoint using UDP.
Jan 17 14:08:20.196 DEBG created UDP  endpoint 0.0.0.0:5683
Registering CoAP resources.
Entering main loop.
Jan 17 14:08:22.758 DEBG ***127.0.0.1:5683 <-> 127.0.0.1:51627 (if1) UDP : new incoming session
Jan 17 14:08:22.758 DEBG *  127.0.0.1:5683 <-> 127.0.0.1:51627 (if1) UDP : received 7 bytes
v:1 t:CON c:GET i:2248 {01} [ Uri-Path:, Uri-Path: ]
Jan 17 14:08:22.758 DEBG request for unknown resource '/', return 4.04
Jan 17 14:08:22.758 DEBG *  127.0.0.1:5683 <-> 127.0.0.1:51627 (if1) UDP : sent 15 bytes
v:1 t:ACK c:4.04 i:2248 {01} [ ] :: 'Not Found'
Jan 17 14:08:27.447 DEBG ***127.0.0.1:5683 <-> 127.0.0.1:53556 (if1) UDP : new incoming session
Jan 17 14:08:27.447 DEBG *  127.0.0.1:5683 <-> 127.0.0.1:53556 (if1) UDP : received 12 bytes
v:1 t:CON c:GET i:abd3 {01} [ Uri-Path:attest ]
Jan 17 14:08:27.447 DEBG call custom handler for resource 'attest'
Received message.
Err: cannot response.
Jan 17 14:08:27.447 DEBG *  127.0.0.1:5683 <-> 127.0.0.1:53556 (if1) UDP : sent 4 bytes
v:1 t:ACK c:0.00 i:abd3 {} [ ] :: 'Response from server.\x0A\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x91\x02\x00\x00\x00\x00\x00\x00\xE0\xEB|\x80`\x7F\x00\x00\xE0\xEB|\x80`\x7F\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00
^CJan 17 14:08:39.509 DEBG ***127.0.0.1:5683 <-> 127.0.0.1:51627 (if1) UDP : session closed
Jan 17 14:08:39.509 DEBG ***127.0.0.1:5683 <-> 127.0.0.1:53556 (if1) UDP : session closed
```

Reply:
Ok. LGTM